franklin/builder/models.py

```diff
@@ -1,33 +1,148 @@
 import json
 import logging
 import os
+import re
 import requests
 from requests.exceptions import ConnectionError, HTTPError, Timeout
 import sys
 import uuid
 
 from django.db import models
+from django.utils import timezone
+from django.utils.translation import ugettext as _
 
 from rest_framework import status
 
 logger = logging.getLogger(__name__)
 
+class Owner(models.Model):
+    """ Represents a github user or organization that owns repos
+
+    :param name: The unique name for this user or org (taken from github)
+    :param github_id: Unique ID github has assigned the owner
+    """
+    name = models.CharField(max_length=100)
+    github_id = models.PositiveIntegerField(unique=True)
+
+    def __str__(self):
+        return self.name
+    
+    class Meta(object):
+        verbose_name = _('Owner')
+        verbose_name_plural = _('Owners')
+
 
 class Site(models.Model):
     """ Represents a 'deployed' or soon-to-be deployed static site.
 
-    :param id: A unique site identified
-    :param owner: The name of the owner of the repo on github. (User or Org)
-    :param owner_id: Unqiue ID github has assigned the owner
-    :param repo_name: The name of the project on github
-    :param repo_name_id: Unique ID github has assigned the project
+    :param owner: Ref to the owner of the project
+    :param name: The name of the project on github
+    :param github_id: Unique ID github has assigned the project
+    :param deploy_key: Token used to access the project on github
+    """
+
+    DEFAULT_ENV = 'Production'
+
+    owner = models.ForeignKey(Owner, related_name='sites')
+    name = models.CharField(max_length=100)
+    github_id = models.PositiveIntegerField(unique=True)
+    deploy_key = models.CharField(max_length=255, default='')
+
+    def get_deployable_event(self, github_event):
```

Comment:
This method does the magic of taking the webhook from github and determining if this project has ANY environments that have a deployment strategy that matches this event. (matching a tag regex or a push on a certain branch)